### ISSUE TITLE
System error #11: Resource temporarily unavailable

### ISSUE THREAD

**Issue by DengHAU** [nanoporetech/medaka#164]
Hi,

I run medaka on a 128 cores CentOS system using `medaka_consensus -i reads/filter.all.fa.gz -d assembly.contig.fasta -o ./ -t 96 -m r941_prom_high_g344`
And I got this errors:
`OMP: Error #34: System unable to allocate necessary resources for OMP thread:
OMP: System error #11: Resource temporarily unavailable
OMP: Hint: Try decreasing the value of OMP_NUM_THREADS.
OMP: Info #242: KMP_AFFINITY: pid 23449 thread 3778 bound to OS proc set {40}
/home/miniconda3/envs/medaka/bin/medaka_consensus: line 134: 23449 Aborted                 (core dumped) medaka consensus ${CALLS2DRAFT}.bam ${CONSENSUSPROBS} --model ${MODEL} --batch_size ${BATCH_SIZE} --threads ${THREADS}
Failed to run medaka consensus.`

Any suggestions?
Thansk!

**Comment by cjw85** [comment]
Hi @DengHAU,

I would first try applying the hint as suggested in the error message above. Before running `medaka_consensus` set the environment variable:

```export OMP_NUM_THREADS=24```

**Comment by DengHAU** [comment]
Thanks for your quickly reply!

what's the meaning 'OMP_NUM_THREADS=24', is this use 24 threads in consensus step? If I want to use all the 128 cores, how can I set the `OMP_NUM_THREADS` par?

Thanks a lot!

**Comment by cjw85** [comment]
Exactly, this tells your computer to use 24 threads for the neural network calculations within medaka. The default for this setting is to use all available cores of you computer; explicitly setting the value to 128 will likely lead to the same error message above.

In addition for the size of calculations that medaka undertakes, more than 24 threads is unlikely to give higher performance.

**Comment by DengHAU** [comment]
OK, got it, Thanks!

**Comment by DengHAU** [comment]
I set `export OMP_NUM_THREADS=24` and reruned medaka, it finished extremly fast just in two seconds! And I got an empty "consensus.fasta"

`Checking program versions
This is medaka 1.0.1
Program    Version    Required   Pass     
bcftools   1.9        1.9        True     
bgzip      1.9        1.9        True     
minimap2   2.17       2.11       True     
samtools   1.9        1.9        True     
tabix      1.9        1.9        True     
Warning: Output ./ already exists, may use old results.
Not aligning basecalls to draft, calls_to_draft.bam exists.
Not running medaka consensus, consensus_probs.hdf exists.
Using medaka stitch to create consensus.
[16:44:57 - DataIndex] Loaded 1/1 (100.00%) sample files.
Polished assembly written to .//consensus.fasta, have a nice day.`

May be someting wrong?
Thanks!

**Comment by cjw85** [comment]
As the message suggests you have rerun medaka using the same output directory as your previous run. You should remove this before rerunning.

**Comment by DengHAU** [comment]
Sorry to bother again. 
I set `OMP_NUM_THREADS=24` and finished consesus step. When `consensus_probs.hdf` generated, medaka failed again because of the same problem!
`[16:47:34 - PWorker] 100.0% Done (2226.1/2226.1 Mbases) in 85497.9s
[16:47:35 - PWorker] All done, 14 remainder regions.
[16:47:41 - Predict] Processing 14 short region(s).
[16:47:41 - ModelLoad] Building model with cudnn optimization: False
[16:47:42 - DLoader] Initializing data loader
[16:47:42 - Sampler] Initializing sampler for consensus of region ctg000011:1901372-1908408.
[16:47:42 - PWorker] Running inference for 0.0M draft bases.
Traceback (most recent call last):
  File "/home/miniconda3/envs/medaka/lib/python3.6/site-packages/h5py/_hl/files.py", line 111, in make_fid
    fid = h5f.open(name, h5f.ACC_RDWR, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 78, in h5py.h5f.open
**OSError: Unable to open file (unable to lock file, errno = 11, error message = 'Resource temporarily unavailable**')`

So I use `medaka stitch consensus_probs.hdf consensus.fasta` to generate consesus fasta file, I also got the same error `OSError: Unable to open file (unable to lock file, errno = 11, error message = 'Resource temporarily unavailable')`

Any suggestions?
Thanks!
